### PR TITLE
boards: acn52832: Add support for acn52832

### PR DESCRIPTION
**Pull request by sven-hm**
This PR adds support for the [ac52832 board](https://aconno.de/products/acn52832/).

### CODE REVIEW

File: boards/arm/acn52832/acn52832.dts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+	compatible = "nordic,nrf-uart";
+	current-speed = <115200>;
+	status = "okay";
+	pinctrl-0 = <&uart0_default>;
+	pinctrl-1 = <&uart0_sleep>;
+	pinctrl-names = "default", "sleep";
+};
+
+&flash0 {
+
```
Comment (anangl) [inline boards/arm/acn52832/acn52832.dts:66]: Nit: unnecessary blank line.

File: boards/arm/acn52832/acn52832.dts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+			reg = <0x0000c000 0x32000>;
+		};
+		slot1_partition: partition@3e000 {
+			label = "image-1";
+			reg = <0x0003e000 0x32000>;
+		};
+		scratch_partition: partition@70000 {
+			label = "image-scratch";
+			reg = <0x00070000 0xa000>;
+		};
```
Comment (anangl) [inline boards/arm/acn52832/acn52832.dts:87]: Perhaps this can be removed. See #52028.

**Comment by sven-hm** [comment]
@anangl thank you for the review! I addressed all your requested changes. If you are ok with the PR now, i would rebase it onto main.

**Review by anangl — COMMENTED** [review]
One address to be corrected. Otherwise, LGTM.

File: boards/arm/acn52832/doc/index.rst [hunk 3]
```diff
@@ -0,0 +1,18 @@
+.. _acn52832:
+
+acn52832
+########
+
+Overview
+********
+
+The acn52832 is a module in a small form factor which features the Nordic Semiconductor
```
Comment (nordicjm) [inline boards/arm/acn52832/doc/index.rst:9]: This documentation is incredibly light, can you look at documentation for other boards and add more to it?

Comment (sven-hm) [reply boards/arm/acn52832/doc/index.rst:9]: I added some more documentation.

**Review by nordicjm — APPROVED** [review]

**Review by anangl — APPROVED** [review]

**Pull request merged**